src/lib/psk.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const Details = require('../utils/details')
+const Transport = require('./transport')
+const cryptoHelper = require('../utils/crypto')
+const assert = require('assert')
+const base64url = require('../utils/base64url')
+
+/**
+  * @module PSK
+  */
+
+/**
+  * Create a payment request using a Pre-Shared Key (PSK).
+  *
+  * @param {Object} params Parameters for creating payment request
+  * @param {String} params.destinationAmount Amount that should arrive in the recipient's account. This value is a string representation of an integer, expressed in the lowest indivisible unit supported by the ledger.
+  * @param {String} params.destinationAccount Target account's ILP address
+  * @param {String} params.sharedSecret Shared secret for PSK protocol
+  * @param {String} [params.id=uuid.v4()] Unique ID for the request (used to ensure conditions are unique per request)
+  * @param {String} [params.expiresAt=30 seconds from now] Expiry of request
+  * @param {Object} [params.data=null] Additional data to include in the request
+  * @param {Object} [params.headers=null] Additional headers for private PSK details
+  * @param {Object} [params.unsafeHeaders=null] Additional headers for public PSK details
+  *
+  * @return {Object} Payment request
+  */
+function createPacketAndCondition (rawParams) {
+  const params = Object.assign({}, rawParams, { secret: rawParams.sharedSecret })
+  return Transport.createPacketAndCondition(params, 'psk')
+}
+
+/**
+  * Generate shared secret for Pre-Shared Key (PSK) transport protocol.
+  *
+  * @param {Object} params Parameters for creating PSK params
+  * @param {String} params.destinationAccount The ILP address that will receive PSK payments
+  * @param {Buffer} params.secretSeed secret used to generate the shared secret and the extra segments of destinationAccount
+  *
+  * @return {PskParams}
+  */
+function generateParams ({
+  destinationAccount,
+  secretSeed
+}) {
+  assert(typeof destinationAccount === 'string', 'destinationAccount must be a string')
+  assert(Buffer.isBuffer(secretSeed), 'secretSeed must be a buffer')
+
+  const token = base64url(cryptoHelper.getPskToken(secretSeed))
+  const sharedSecret =
+    base64url(cryptoHelper.getPskSharedSecret(secretSeed, token))
+
+  const receiverId = base64url(cryptoHelper.getReceiverId(sharedSecret))
```

Comment:
I think lines 49 - 53 should be moved into the crypto helper, which could just export a function `generatePskParams(secretSeed) => {sharedSecret, receiverId, token}` or something like that. I think that would make some of the crypto easier to follow